src/main/C594C.c

```diff
@@ -108,17 +120,100 @@ INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D6700);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D6780);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D67A0);
+void func_800D67A0(struct QuadObj* arg0)
+{
+    D_8010FCAC[arg0->state](arg0);
+}
 
-INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D67DC);
+void func_800D67DC(struct QuadObj* arg0)
+{
+    struct PlayerObj* ptr = &D_801418C8;
+    register struct Unk* temp_a2 asm("a2") = arg0->unk5C; // fake
+    s32 var_a3 = 0;
+    if (D_8014198B != 0) {
+        var_a3 = 1;
+    }
+    if (D_8014198C != 0) {
+        var_a3 = 1;
+    }
+    if (D_8014195B != 6) {
+        var_a3 = 1;
+    }
+    if (D_80141924 == 0) {
+        var_a3 = 1;
+    }
+    if (D_80141987 != 0) {
+        var_a3 = 1;
+    }
+    if (arg0->unk5C->base.active == 0) {
+        var_a3 = 1;
+    }
+    if (arg0->unk5C->unk5C == 0) {
+        var_a3 = 1;
+    }
+    if (var_a3 != 0) {
+        ZeroObjectState(arg0);
+        return;
+    }
+    if (arg0->state == 0) {
+        func_800D68D0(arg0, ptr);
+        return;
+    }
+    func_800D6944(arg0, ptr);
+}
 
-INCLUDE_ASM("asm/us/main/nonmatchings/C594C", func_800D68D0);
+void func_800D68D0(struct QuadObj* arg0, struct PlayerObj* arg1)
+{
+    arg0->active = -0x7D;
+    arg0->unk3 = 1;
+    arg0->unk37 = arg1->base.unk14;
+    arg0->unk36 = 1;
+    *(s8*)&arg0->unk38 = 0x3C;
```

Comment:
Wondering if this is the equivalent of the `.ext` in sotn that's variable per-object. This is fine for now